internal/astutils/ast_utils.go

```diff
@@ -15,28 +15,34 @@ func FuncSignatureIs(funcDecl *ast.FuncDecl, wantName string, wantParametersType
 		return false // func name doesn't match expected one
 	}
 
-	funcParametersTypes := getTypeNames(funcDecl.Type.Params)
-	if len(wantParametersTypes) != len(funcParametersTypes) {
-		return false // func has not the expected number of parameters
-	}
-
 	funcResultsTypes := getTypeNames(funcDecl.Type.Results)
 	if len(wantResultsTypes) != len(funcResultsTypes) {
 		return false // func has not the expected number of return values
 	}
 
-	for i, wantType := range wantParametersTypes {
-		if wantType != funcParametersTypes[i] {
-			return false // type of a func's parameter does not match the type of the corresponding expected parameter
-		}
-	}
-
 	for i, wantType := range wantResultsTypes {
 		if wantType != funcResultsTypes[i] {
 			return false // type of a func's return value does not match the type of the corresponding expected return value
 		}
 	}
 
+	// Name and return values are those we expected, the final result depends on parameters being what we want.
+	return FuncParametersSignatureIs(funcDecl, wantParametersTypes)
+}
+
+// FuncParametersSignatureIs returns true if the function has parameters of the given type and order, false otherwise
+func FuncParametersSignatureIs(funcDecl *ast.FuncDecl, wantParametersTypes []string) bool {
+	funcParametersTypes := getTypeNames(funcDecl.Type.Params)
+	if len(wantParametersTypes) != len(funcParametersTypes) {
+		return false // func has not the expected number of parameters
+	}
+
+	for i, wantType := range wantParametersTypes {
+		if wantType != funcParametersTypes[i] {
+			return false // type of a func's parameter does not match the type of the corresponding expected parameter
+		}
+	}
+
 	return true
```

Comment:
I had to read it multiple times to understand what you try to achieve.

Isn't simply about using Go std [slices.Equal](https://pkg.go.dev/slices#Equal) ?


```suggestion
	funcParametersTypes := getTypeNames(funcDecl.Type.Params)

 	return slices.Equal(wantParametersTypes, funcParametersTypes)
```